Library/Homebrew/cmd/update-bash.sh

```diff
@@ -320,8 +333,23 @@ EOS
   # this procedure will be removed in the future if it seems unnecessary
   rename_taps_dir_if_necessary
 
-  # kill all of subprocess on interrupt
-  trap '{ pkill -P $$; wait; exit 130; }' SIGINT
+  # Spawn ssh-agent if it's not running and at least one of repos is using ssh
+  # https://github.com/Homebrew/homebrew/issues/48267
+  for DIR in "$HOMEBREW_REPOSITORY" "$HOMEBREW_LIBRARY"/Taps/*/*
+  do
+    [[ -d "$DIR/.git" ]] || continue
+    cd "$DIR" || continue
+    [[ "$(git config --get remote.origin.url 2>/dev/null)" =~ ^https?:// ]] && continue
+    if ! [[ -n "$SSH_AUTH_SOCK" && -e "$SSH_AUTH_SOCK" ]]
+    then
+      echo "starting ssh-agent..."
```

Comment:
I think we should output unconditionally since we are asking password here. And these part of logics shouldn't  be run very often.